src/UCTNode.cpp

```diff
@@ -112,17 +112,26 @@ bool UCTNode::create_children(std::atomic<int> & nodecount,
     FastBoard & board = state.board;
     std::vector<Network::scored_node> nodelist;
 
+    auto legal_sum = 0.0f;
     for (auto& node : raw_netlist.first) {
         auto vertex = node.second;
         if (vertex != FastBoard::PASS) {
             if (vertex != state.m_komove
                 && !board.is_suicide(vertex, board.get_to_move())) {
                 nodelist.emplace_back(node);
+                legal_sum += node.first;
             }
         } else {
             nodelist.emplace_back(node);
+            legal_sum += node.first;
         }
     }
+
+    // re-normalize after removing illegal moves.
+    for (auto& node : nodelist) {
+        node.first /= legal_sum;
```

Comment:
In theory this shouldn't be zero as this loop won't run if there are no legal moves, but if the policy prior value for a move underflows, you might still end up dividing by zero. This bit us in the dirichlet noise generation. So I'd add the same guard.

Reply:
Fixed.